zkvm/src/vm.rs

```diff
@@ -567,6 +573,33 @@ where
         })
     }
 
+    fn delegate(&mut self) -> Result<(), VMError> {
+        // Pop signature and put into signature struct
+        let sig = self.pop_item()?.to_data()?.to_bytes();
+        if sig.len() != 64 {
+            return Err(VMError::FormatError);
+        }
+        let mut buf = [0u8; 64];
+        buf[..].copy_from_slice(&sig);
+        let signature = Signature::from_bytes(buf)?;
```

Comment:
This code does not check that `sig` is fully read.

Let's add `read_u8x64` method to `encoding::SliceReader` and do `Signature::from_bytes(SliceReader::parse(&sig){|r| r.read_u8x64() }?)`.